sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py

```diff
@@ -0,0 +1,210 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from logging import getLogger
+from typing import Any, Dict
+
+from azure.monitor.opentelemetry._constants import (
+    DISABLE_LOGGING_ARG,
+    DISABLE_METRICS_ARG,
+    DISABLE_TRACING_ARG,
+    EXCLUDE_INSTRUMENTATIONS_ARG,
+    INSTRUMENTATION_CONFIG_ARG,
+    LOGGER_NAME_ARG,
+    LOGGING_EXPORT_INTERVAL_MS_ARG,
+    LOGGING_LEVEL_ARG,
+    METRIC_READERS_ARG,
+    RESOURCE_ARG,
+    SAMPLING_RATIO_ARG,
+    TRACING_EXPORT_INTERVAL_MS_ARG,
+    VIEWS_ARG,
+)
+from azure.monitor.opentelemetry._types import ConfigurationValue
+from azure.monitor.opentelemetry.exporter import (
+    ApplicationInsightsSampler,
+    AzureMonitorLogExporter,
+    AzureMonitorMetricExporter,
+    AzureMonitorTraceExporter,
+)
+from azure.monitor.opentelemetry.util.configurations import _get_configurations
+from opentelemetry._logs import get_logger_provider, set_logger_provider
+from opentelemetry.instrumentation.dependencies import (
+    get_dist_dependency_conflicts,
+)
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.metrics import set_meter_provider
+from opentelemetry.sdk._logs import LoggerProvider, LoggingHandler
+from opentelemetry.sdk._logs.export import BatchLogRecordProcessor
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.trace import get_tracer_provider, set_tracer_provider
+from pkg_resources import iter_entry_points
+
+_logger = getLogger(__name__)
+
+
+_SUPPORTED_INSTRUMENTED_LIBRARIES = (
+    "django",
+    "fastapi",
+    "flask",
+    "psycopg2",
+    "requests",
+    "urllib",
+    "urllib3",
+)
+
+
+InstrumentationConfig = Dict[str, Dict[str, Any]]
+
+
+def configure_azure_monitor(**kwargs) -> None:
```

Comment:
@annatisch You mentioned something about changing the way we use kwargs but I did not understand it.

Reply:
Yes, this method signature should use python 3 syntax rather than `**kwargs`, so something like:
```python
def configure_azure_monitor(
    *,
    connection_string: str,
    disable_logging: bool = False,
    exclude_instrumentations: Optional[Sequence[str]] = None,
    **kwargs
)
```

Reply:
This would have to look something like this:
```python
def configure_azure_monitor(
        *,
        connection_string: str = None,
        **kwargs
    ) -> None:
```
... or, even we also need to include pass through kwargs in the method signature:
```python
def configure_azure_monitor(
        *,
        connection_string: str = None,
        credential: Union[ManagedIdentityCredential, ClientSecretCredential] = None,
        disable_offline_storage: bool = None,
        storage_directory: str = None,
        **kwargs
    ) -> None:
```
At this point, the kwargs don't actually do anything. For instance, the kwargs no longer includeany supported arguments. As such, the configuration layer would have to be rewritten. In my view, since these arguments are primarily pass throughs to the exporter, using kwargs like how the function is currently written best represents the functionality.